script.js

```diff
@@ -22,59 +22,58 @@ xmlhttp.onreadystatechange = function() {
 xmlhttp.open("GET", "data/data.json", true);
 xmlhttp.send();
 
-function generateDropdown(){
+const generateDropdown = () => {
   console.log('generating dropdown');
   let optionsHTML = "";
   leftEmoji = data.groups[0].options[0];
   rightEmoji = data.groups[0].options[0];
-  for(let i = 0; i < data.groups.length; i++){
-    let currentGroup = data.groups[i];
-    optionsHTML = optionsHTML + '<optgroup label="' + currentGroup.label + '">';
-    for(let j = 0; j < currentGroup.options.length; j++){
-      console.log(currentGroup.options[j]);
-      optionsHTML = optionsHTML + '<option>' + currentGroup.options[j].emoji + ' ' + currentGroup.options[j].label + '</option>';
+  for (const group of data.groups) {
```

Comment:
`for...of` is better than `for...in` in pure js, but `for of` is much legit. I think if you use eslint, probably it will recommend you to use `for...of`